src/syscheckd/fim_db.c

```diff
@@ -728,7 +735,7 @@ int fim_db_data_checksum_range(fdb_t *fim_sql, const char *start, const char *to
             goto end1;
         }
         entry = fim_db_decode_full_row(fim_sql->stmt[FIMDB_STMT_GET_PATH_RANGE]);
-        if (i == m) {
+        if (i == m && entry->path) {
```

Comment:
In the function `fim_db_decode_full_row()` please, change the use of `w_strdup()` to `os_strdup()`.